app/controllers/CyaUpdateRecordController.scala

```diff
@@ -171,49 +177,67 @@ class CyaUpdateRecordController @Inject() (
             _                        <- sessionRepository.set(updatedAnswers)
           } yield Redirect(routes.SingleRecordController.onPageLoad(recordId))
-        case Left(errors) => Future.successful(logErrorsAndContinue(errors, recordId))
+        case Left(errors) =>
+          Future.successful(
+            logErrorsAndContinue(errors, routes.CyaUpdateRecordController.onPageLoadCountryOfOrigin(recordId))
+          )
       }
     }
 
   def onSubmitGoodsDescription(recordId: String): Action[AnyContent] =
     (identify andThen getData andThen requireData).async { implicit request =>
-      UpdateGoodsRecord.buildGoodsDescription(request.userAnswers, request.eori, recordId) match {
-        case Right(model) =>
-          auditService.auditFinishUpdateGoodsRecord(recordId, request.affinityGroup, model)
+      UpdateGoodsRecord.buildGoodsDescription(request.userAnswers, recordId) match {
+        case Right(goodsDescription) =>
+          auditService.auditFinishUpdateGoodsRecord(
+            recordId,
+            request.affinityGroup,
+            UpdateGoodsRecord(request.eori, recordId, goodsDescription = Some(goodsDescription))
+          )
           for {
-            _                        <- goodsRecordConnector.updateGoodsRecord(model)
+            _                        <- goodsRecordConnector.updateGoodsRecord(
+                                          UpdateGoodsRecord(request.eori, recordId, goodsDescription = Some(goodsDescription))
+                                        )
             updatedAnswersWithChange <-
               Future.fromTry(request.userAnswers.remove(HasGoodsDescriptionChangePage(recordId)))
             updatedAnswers           <- Future.fromTry(updatedAnswersWithChange.remove(GoodsDescriptionUpdatePage(recordId)))
             _                        <- sessionRepository.set(updatedAnswers)
           } yield Redirect(routes.SingleRecordController.onPageLoad(recordId))
-        case Left(errors) => Future.successful(logErrorsAndContinue(errors, recordId))
+        case Left(errors)            =>
+          Future.successful(
+            logErrorsAndContinue(errors, routes.CyaUpdateRecordController.onPageLoadGoodsDescription(recordId))
+          )
       }
     }
 
   def onSubmitCommodityCode(recordId: String): Action[AnyContent] =
     (identify andThen getData andThen requireData).async { implicit request =>
-      UpdateGoodsRecord.buildCommodityCode(request.userAnswers, request.eori, recordId) match {
-        case Right(model) =>
-          auditService.auditFinishUpdateGoodsRecord(recordId, request.affinityGroup, model)
+      UpdateGoodsRecord.buildCommodityCode(request.userAnswers, recordId) match {
+        case Right(commodity) =>
+          auditService.auditFinishUpdateGoodsRecord(
+            recordId,
+            request.affinityGroup,
+            UpdateGoodsRecord(request.eori, recordId, commodityCode = Some(commodity), category = Some(1))
```

Comment:
Why do we always set category to 1. are there other options? Apologies I am missing context here :) same below for line 223

Reply:
Yes. It should be `category = Some(1)` https://github.com/hmrc/trader-goods-profiles-frontend/pull/290/files#diff-0d95f1a32505dd8cdc88e6597445c3bcebc79eaf8d2832c9cb02b26466b5cd60R53